src/react-components/auth/SignInModal.js

```diff
@@ -0,0 +1,134 @@
+import React, { useCallback, useState } from "react";
+import PropTypes from "prop-types";
+import { CloseButton } from "../input/CloseButton";
+import { Modal } from "../modal/Modal";
+import { FormattedMessage } from "react-intl";
+import styles from "./SignInModal.scss";
+import { Button } from "../input/Button";
+import { TextInputField } from "../input/TextInputField";
+
+export const SignInStep = {
+  submit: "submit",
+  waitForVerification: "waitForVerification",
+  complete: "complete"
+};
+
+export function SubmitEmail({ onSubmitEmail, initialEmail, showPrivacy, privacyUrl, showTerms, termsUrl, message }) {
+  const [email, setEmail] = useState(initialEmail);
+
+  const onSubmitForm = useCallback(
+    e => {
+      e.preventDefault();
+      onSubmitEmail(email);
+    },
+    [onSubmitEmail, email]
+  );
+
+  return (
+    <form onSubmit={onSubmitForm} className={styles.modalContent}>
+      <p>{message || <FormattedMessage id="sign-in.prompt" />}</p>
+      <TextInputField
+        name="email"
+        type="email"
+        required
+        value={email}
+        onChange={e => setEmail(e.target.value)}
```

Comment:
Yeah, so in this case, it's not great but it's not terrible. The main perf consideration is not GC or the overhead of redefining the function every render. The fact that the arrow function is redefined every render makes it so that the TextInputField will also re-render since the props changed. This really shouldn't be an issue, but it's easy to fix with `useCallback`. I want to go back and fix up perf everywhere by using `useCallback`, `useMemo`, and `memo` to ensure that components only re-render when their props changed. However, unless it's happening often, it's really not a big deal.